Jenkinsfile

```diff
@@ -401,66 +401,6 @@ pipeline {
                 expression { !skipStage() }
             }
             parallel {
-                stage('checkpatch') {
-                    when {
-                        beforeAgent true
-                        expression { !skipStage() }
-                    }
-                    agent {
-                        dockerfile {
-                            filename 'Dockerfile.checkpatch'
-                            dir 'utils/docker'
-                            label 'docker_runner'
-                            additionalBuildArgs dockerBuildArgs(add_repos: false)
-                        }
-                    }
-                    steps {
-                        checkPatch user: GITHUB_USER_USR,
```

Comment:
Are we really ready to remove checkpatch entirely just be putting shellcheck into GHA?

Reply:
yes!

Reply:
I think it's great to do this, but will it introduce more disparity between the master and 2.4 checks if not backported?

Reply:
That's inevitable.

We will have a master that is shellcheck-clean but release branches that still has issues.

Reply:
There are two models for dealing with warnings, either we accept we have them and warn but do not fail PRs where files with warnings are modified.  That's what we used to do with pylint and we do today with shellcheck on both master and the release branch.  The other model is the tree is clean, we check the whole tree on every PR and fail the build on warrnings.

At this point there is always going to be a release build, to not have disparity between branches then we'd have to fix all the issues on master, fix all the issue on the release branch, then upgrade the test on master and then release in the mean time not landing any regressions to master.

Clearly possible but a lot more work and for the small number of files affected is it worth it?

Shellcheck is also the last check (that I'm aware of) where we only check the diff, other than the clang-format change which is a lot newer and and differs there because most code isn't compliant.